src/phoenix/server/api/types/Project.py

```diff
@@ -142,14 +142,24 @@ async def latency_ms_quantile(
         time_range: Optional[TimeRange] = UNSET,
     ) -> Optional[float]:
         return await info.context.data_loaders.latency_ms_quantile.load(
-            (self.name, time_range, probability)
+            (self.id_attr, time_range, probability)
         )
 
     @strawberry.field
-    def trace(self, trace_id: ID) -> Optional[Trace]:
-        if self.project.has_trace(TraceID(trace_id)):
-            return Trace(trace_id=trace_id, project=self.project)
-        return None
+    async def trace(self, info: Info[Context, None], trace_id: ID) -> Optional[Trace]:
+        async with info.context.db() as session:
+            if not await session.scalar(
+                select(1)
+                .join_from(models.Trace, models.Project)
```

Comment:
same

Reply:
Do we still need the projects table here?

Reply:
removed